NWNXLib/Assert.hpp

```diff
@@ -31,13 +31,21 @@ namespace Assert {
     #define ASSERT_FAIL_MSG(format, ...) (void)0
 #endif
 
+    #define ASSERT_THROW(condition) \
+        do \
+        { \
+            if(!(condition)) ::NWNXLib::Assert::Throw((#condition)); \
```

Comment:
Just inline it directly. The function isn't adding anything useful, just an extra indirection step, and an extra step that has to be unwound when throwing.

Reply:
Inline'd! If this is good I'll start replacing all the `ASSERT()`s

Reply:
Yes, very good! A most _minor_ nit is that I'd prefer the name `ASSERT_OR_THROW`, but I don't feel that strongly. Whatever you like better.